init_test.go

```diff
@@ -163,15 +163,15 @@ func TestHandleInitRes(t *testing.T) {
 	listenerComplete := make(chan struct{})
 
 	go func() {
-		conn, err := l.Accept()
-		require.NoError(t, err, "l.Accept failed")
+		conn, err2 := l.Accept()
```

Comment:
Prefer not to name this `err2`, I think instead of making this change, the best option might be to have a helper,
```go

func newListener(t *testing.T) net.Listener {
  l, err := net.Listen("tcp", ":0")
  require.NoError(t, err, "Listen failed")
  return l
}
```

That way we don't create a variable `err` before starting the goroutine, and should avoid the shadowing error.